### PR TITLE
Use separate undefined object for else-less conditional expressions

### LINKED ISSUE #710: `{{ 'foo' if false }}` shortcut not usable with StrictUndefined
Using the shortcut version of the ternary operator with no `else` implies `else <undefined>`. This is fine when using the normal `Undefined` but makes this shortcut basically useless with `StrictUndefined` (and to some extent `LoggingUndefined`).

```python
>>> from jinja2 import StrictUndefined, Environment
>>> env = Environment(undefined=StrictUndefined)
>>> env.from_string('{{ "foo" if false }}').render()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/adrian/.python-env/lib/python2.7/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/adrian/.python-env/lib/python2.7/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "<template>", line 1, in top-level template code
jinja2.exceptions.UndefinedError: the inline if-expression on line 1 evaluated to false and no else section was defined.
```

I'd change the default from `<undefined>` to `''` - like this the behavior of existing templates will not change (empty output) but it would work fine with the stricter `Undefined` variants.

The only breakage I could imagine is someone using it in a function/macro call with an explicit `is defined` check later on. But I think that's rather unlikely and if necessary a new policy option could be added to make it default to `<undefined>` again.

### PR DESCRIPTION
**Pull request by ThiefMaster**
fixes #710

### CODE REVIEW

File: jinja2/environment.py [hunk 1]
```diff
@@ -283,7 +293,8 @@ def __init__(self,
                  cache_size=400,
                  auto_reload=True,
                  bytecode_cache=None,
-                 enable_async=False):
+                 enable_async=False,
+                 cond_expr_undefined=Undefined):
```
Comment (ThiefMaster) [inline jinja2/environment.py]: For more backwards compatibility this could be set to `None` here and use the value of `undefined` in that case.

**Comment by davidism** [comment]
Do we need this to be configurable? It seems like the contract of `value if condition` is that it returns `Undefined`, maybe it can just be that simple.

**Comment by ThiefMaster** [comment]
That was my question ;) Then I'll change the default to use the `undefined` value if nothing is specified.

**Comment by ThiefMaster** [comment]
updated

**Comment by ThiefMaster** [comment]
oh, I think I misread your comment, i thought you mean that it's supposed to return the `undefined` object by default (which is whatever is configured in the env as `undefined`)

**Pull request merged**